app/app/src/main/java/de/stustapay/stustapay/ui/ticket/TicketViewModel.kt

```diff
@@ -262,30 +261,10 @@ class TicketViewModel @Inject constructor(
     }
 
     private suspend fun bookSale(paymentMethod: PaymentMethod) {
-        _saleCompleted.update { null }
-
-        val response = ticketRepository.bookTicketSale(
-            _ticketDraft.value.getNewTicketSale(paymentMethod)
-        )
-
-        when (response) {
-            is Response.OK -> {
-                // delete the sale draft
-                clearDraft()
-                _status.update { resourcesProvider.getString(R.string.ticket_order_booked) }
-                // now we have a completed sale
-                _saleCompleted.update { response.data }
-                _navState.update { TicketPage.Done }
-            }
+        infallible.bookTicketSale(_ticketDraft.value.getNewTicketSale(paymentMethod))
 
-            is Response.Error.Service -> {
-                _navState.update { TicketPage.Error }
-                _status.update { response.msg() }
-            }
-
-            is Response.Error -> {
-                _status.update { response.msg() }
-            }
-        }
+        clearDraft()
+        _status.update { resourcesProvider.getString(R.string.ticket_order_booked) }
+        _navState.update { TicketPage.Done }
```

Comment:
we should relay and display actual error messages, i.e. when the submission to the server was then done

Reply:
I wasn't able to think of a good way to generically return data from infallible requests. The requirement for serialisation prevents us from using pretty much all good tools for this, such as Flows, lambdas, suspend fns. Do you have any ideas for approaches that I could try?
However, the backend can't actually return errors at this point by definition, so its reasonable to assume that the request was successful.

Reply:
the sane theoretical way to solve this is to have an persistent action queue, where tasks are retried and are pending until a result was retrieved. the status of pending task(s) should then be displayed so that the general case looks and behaves nice. and i wouldn't enqueue tasks that make no sense queuing, i.e. telling a customer their payment worked.